receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -0,0 +1,224 @@
+
+	server  *http.Server
+	config  *Config
+	logger  *zap.Logger
+	obsrecv *obsreport.Receiver
+}
+
+// NewPrometheusRemoteWriteReceiver - remote write
+func NewPrometheusRemoteWriteReceiver(config *Config, consumer consumer.Metrics, params component.ReceiverCreateSettings) (*PrometheusRemoteWriteReceiver, error) {
+	zr := &PrometheusRemoteWriteReceiver{
+		params:       params,
+		nextConsumer: consumer,
+		config:       config,
+		logger:       params.Logger,
+		obsrecv: obsreport.NewReceiver(obsreport.ReceiverSettings{
+			ReceiverID:             config.ID(),
+			ReceiverCreateSettings: params,
+		}),
+	}
+	return zr, nil
+}
+
+// Start - remote write
+func (rec *PrometheusRemoteWriteReceiver) Start(ctx context.Context, host component.Host) error {
+	if host == nil {
+		return errors.New("nil host")
+	}
+	rec.mu.Lock()
+	defer rec.mu.Unlock()
+	var err = componenterror.ErrNilNextConsumer
+	rec.startOnce.Do(func() {
+		err = nil
+		rec.host = host
+		rec.server, err = rec.config.HTTPServerSettings.ToServer(host, rec.params.TelemetrySettings, rec)
+		var listener net.Listener
+		listener, err = rec.config.HTTPServerSettings.ToListener()
+		if err != nil {
+			return
+		}
+		rec.shutdownWG.Add(1)
+		go func() {
+			defer rec.shutdownWG.Done()
+			if errHTTP := rec.server.Serve(listener); errHTTP != http.ErrServerClosed {
+				host.ReportFatalError(errHTTP)
+			}
+		}()
+	})
+	return err
+}
+
+func (rec *PrometheusRemoteWriteReceiver) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	req, err := remote.DecodeWriteRequest(r.Body)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	ctx := rec.obsrecv.StartMetricsOp(context.Background())
+
+	reg := regexp.MustCompile(`(\w+)_(\w+)_(\w+)\z`)
+	pms := pmetric.NewMetrics()
+
+	for _, ts := range req.Timeseries {
+
+		prm := pmetric.NewResourceMetrics()
+
+		metricName := finaName(ts.Labels)
+		pm := pmetric.NewMetric()
+		pm.SetName(metricName)
+		rec.logger.Debug("Metric name", zap.String("metric_name", pm.Name()))
+
+		match := reg.FindStringSubmatch(metricName)
+		metricsType := ""
+		if len(match) > 1 {
+			lastSuffixInMetricName := match[len(match)-1]
+			if IsValidSuffix(lastSuffixInMetricName) {
+				metricsType = lastSuffixInMetricName
+				if len(match) > 2 {
+					secondSuffixInMetricName := match[len(match)-2]
+					if IsValidUnit(secondSuffixInMetricName) {
+						pm.SetUnit(secondSuffixInMetricName)
+					}
+				}
+			} else if IsValidUnit(lastSuffixInMetricName) {
+				pm.SetUnit(lastSuffixInMetricName)
+			}
+		}
+		rec.logger.Debug("Metric unit", zap.String("metric name", pm.Name()), zap.String("metric_unit", pm.Unit()))
+
+		for _, s := range ts.Samples {
+			ppoint := pmetric.NewNumberDataPoint()
+			ppoint.SetDoubleVal(s.Value)
+			ppoint.SetTimestamp(pcommon.NewTimestampFromTime(time.Unix(0, s.Timestamp*int64(time.Millisecond))))
+			for _, l := range ts.Labels {
+				ppoint.Attributes().Insert(l.Name, pcommon.NewValueString(l.Value))
```

Comment:
Rename '__name__' label due to conflict between attributes process and Google exporter